pkg/agent/memberlist/cluster.go

```diff
@@ -252,11 +252,14 @@ func (c *Cluster) enqueueExternalIPPool(obj interface{}) {
 // newClusterMember gets the Node's IP and returns a cluster member "<IP>:<clusterMemberlistPort>"
 // representing that Node in the memberlist cluster.
 func (c *Cluster) newClusterMember(node *corev1.Node) (string, error) {
-	nodeAddr, err := k8s.GetNodeAddr(node)
+	nodeAddrs, err := k8s.GetNodeAddrs(node)
 	if err != nil {
-		return "", fmt.Errorf("obtain IP address from K8s Node failed: %v", err)
+		return "", fmt.Errorf("obtain IP addresses from K8s Node failed: %v", err)
 	}
-	member := fmt.Sprintf("%s:%d", nodeAddr, c.bindPort)
+	if nodeAddrs.IPv4 == nil {
+		return "", fmt.Errorf("there is no IPv4 address from K8s Node")
+	}
+	member := fmt.Sprintf("%s:%d", nodeAddrs.IPv4, c.bindPort)
```

Comment:
@wenqiq I think you will need to change this after the PR is merged to support single IPv6 stack case.

Reply:
I think this PR can keep egress feature work in IPv6 cluster easily if it just handles the case like ipsec:
```
nodeAddr := nodeAddrs.IPv4
if nodeAddr == nil {
    nodeAddr = nodeAddrs.IPv6
}
member := fmt.Sprintf("%s:%d", nodeAddr, c.bindPort)
```